### PR TITLE
fix(security): verifier contract range limit checks

### PR DESCRIPTION
**Pull request by skarred14**
# Description
Following an issue raised by @weijiekoh this PR checks for the range limits for proof and public inputs to the `verify()` in `Verifier.sol`. Although there are parameter in the `radish34/api/src/utils/crypto/ecc/babyjubjubparams.js` that are utilized in signature verification, there is currently no additional check against the inputs that any user could enter while interacting directly with the deployed `Verifier` contract

## Related Issue

#34 

## Motivation and Context

Test data for the `Verifier.test.js` is based on running/debugging the integration test

https://github.com/appliedzkp/semaphore/issues/16
https://github.com/appliedzkp/semaphore/blob/master/contracts/sol/verifier.sol#L199

## How Has This Been Tested

All tests at the root level as indicated in `npm run test` pass. New tests at radish34/contracts/ level also pass.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: radish34/contracts/contracts/Verifier.sol [hunk 1]
```diff
@@ -36,6 +36,11 @@ contract Verifier is Ownable, ERC165Compatible, Registrar, IVerifier {
 
     using Pairing for *;
 
+    uint256 constant SNARK_SCALAR_BOUND = 21888242871839275222246405745257275088548364400416034343698204186575808495617;
```
Comment (skarred14) [inline radish34/contracts/contracts/Verifier.sol:39]: @weijiekoh, comments?

**Comment by skarred14** [comment]
test cases updated

File: radish34/contracts/package.json [hunk 2]
```diff
@@ -14,11 +14,17 @@
...
   },
   "dependencies": {
     "@babel/preset-env": "^7.9.0",
+    "big-integer": "^1.6.48",
+    "bignumber": "^1.1.0",
+    "ethereum-doppelganger": "^0.0.7",
+    "ethers": "^4.0.46",
     "openzeppelin-solidity": "^2.5.0"
   },
   "jest": {
```
Comment (bitwiseguy) [inline radish34/contracts/package.json:30]: To get rid of the following error...
```
 FAIL  __tests__/mockVerifierData.js
  ● Test suite failed to run
    Your test suite must contain at least one test.
```
...modify your `jest` block to this:
```
  "jest": {
    "testEnvironment": "node",
    "verbose": true,
    "testPathIgnorePatterns": [
      "./__tests__/mockVerifierData.js"
    ]
  }
```

File: radish34/zkp/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
   "scripts": {
     "build": "babel src -d dist",
     "start": "npm run build && node ./dist/index.js",
-    "dev": "npm run build && nodemon --exec babel-node ./src/index.js",
+    "dev": "npm run start && nodemon --exec babel-node ./src/index.js",
```
Comment (bitwiseguy) [inline radish34/zkp/package.json:18]: I think we should leave this as:
```
"dev": "npm run build && nodemon --exec babel-node ./src/index.js",
```
I don't think there is a need to start a `node` and a `nodemon` process. Should be one or the other

File: radish34/api/package.json [hunk 4]
```diff
@@ -17,7 +17,7 @@
   "scripts": {
     "build": "babel src -d dist",
     "start": "npm run build && node ./dist/index.js",
-    "dev": "npm run build && nodemon --exec babel-node ./src/index.js",
+    "dev": "npm run start && nodemon --exec babel-node ./src/index.js",
```
Comment (bitwiseguy) [inline radish34/api/package.json:20]: I think we should leave this as:
```
"dev": "npm run build && nodemon --exec babel-node ./src/index.js",
```
I don't think there is a need to start a `node` and a `nodemon` process. Should be one or the other

File: radish34/contracts/contracts/Verifier.sol (hunk 1 above)
Comment (weijiekoh) [reply radish34/contracts/contracts/Verifier.sol:39]: Looks good!

**Pull request merged**